src/adagrams.js

```diff
@@ -1,15 +1,229 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = [
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "B",
+    "B",
+    "C",
+    "C",
+    "D",
+    "D",
+    "D",
+    "D",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "F",
+    "F",
+    "G",
+    "G",
+    "G",
+    "H",
+    "H",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "J",
+    "K",
+    "L",
+    "L",
+    "L",
+    "L",
+    "M",
+    "M",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "P",
+    "P",
+    "Q",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "S",
+    "S",
+    "S",
+    "S",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "U",
+    "U",
+    "U",
+    "U",
+    "V",
+    "V",
+    "W",
+    "W",
+    "X",
+    "Y",
+    "Y",
+    "Z",
+  ];
+
+  const shuffledLetterPool = letterPool.sort(() => Math.random() - 0.5);
+
+  let hand = shuffledLetterPool.slice(0, 10);
+
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let loop_count = 0;
+  for (let letter of input) {
+    for (let handLetter of lettersInHand) {
+      if (letter.toUpperCase() == handLetter.toUpperCase()) {
+        loop_count++;
+        const index = lettersInHand.indexOf(handLetter);
+        if (index > -1) {
+          lettersInHand.splice(index, 1);
+        }
+      }
+    }
+  }
+  if (loop_count === input.length) {
+    return true;
+  } else {
+    return false;
+  }
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  let score = 0;
+
+  if (word.length > 6 && word.length < 11) {
+    score += 8;
+  }
+
+  for (let letterCaseSensitive of word) {
+    const letter = letterCaseSensitive.toUpperCase();
+    if (
+      letter === "A" ||
+      letter === "E" ||
+      letter === "I" ||
+      letter === "O" ||
+      letter === "U" ||
+      letter === "L" ||
+      letter === "N" ||
+      letter === "R" ||
+      letter === "S" ||
+      letter === "T"
+    ) {
+      score += 1;
+    } else if (letter === "D" || letter === "G") {
+      score += 2;
+    } else if (
+      letter === "B" ||
+      letter === "C" ||
+      letter === "M" ||
+      letter === "P"
+    ) {
+      score += 3;
+    } else if (
+      letter === "F" ||
+      letter === "H" ||
+      letter === "V" ||
+      letter === "W" ||
+      letter === "Y"
+    ) {
+      score += 4;
+    } else if (letter === "K") {
+      score += 5;
+    } else if (letter === "J" || letter === "X") {
+      score += 8;
+    } else if (letter === "Q" || letter === "Z") {
+      score += 10;
+    }
+  }
+
+  return score;
 };
 
+console.log(scoreWord("A"));
```

Comment:
Logging to the console is a very useful debugging technique, however, we don't want to end up merging code in that contains log statements, so be sure to check your code before submitting a PR to make sure you've removed any debugging console logs.